### DISCUSSION TITLE
Serial Baud 5800 is supported?

### DISCUSSION THREAD

**Discussion by luixxx** [Pi4J/pi4j#140] [Q&A] [answered]
Hi, 
I need to use serial0 port with baud of 5800, 
I get error: PIGPIO ERROR: PI_BAD_SER_SPEED; bas serial baud rate

It is not supported?, 
It will be suported?

Thankyou
Luis

**Comment by FDelporte** [comment]
Hey Luis, this seems to be a very "strange" speed and didn't find it documented somewhere immediately...
See for instance https://tldp.org/HOWTO/Remote-Serial-Console-HOWTO/preparation-setspeed.html

These are the available ones in the library: https://github.com/Pi4J/pi4j-v2/blob/develop/pi4j-core/src/main/java/com/pi4j/io/serial/Baud.java

**Comment by savageautomate** [comment, marked as answer]
I remember looking into this issue several years ago and "_non-standard_" baud rates are not well supported by the Linux UART driver or the RaspberryPi.  Most examples I found were using modified clock rates and C code to pull off a "_non-standard_" baud rate.  

To further complicate the issue, it appears that the PiGPIO library only supports the standard baud rates.  

So, "no" ... at this time the Pi4J project is not able to support non-standard baud rates.  Unless both the underlying drivers and PiGPIO  add support, it's unlikely that we can do much about it. 

I suppose if you had a USB to serial device with its own driver that could support the "_non-standard_" baud rate , then it could be possible with the linux file system serial provider, but we have not implemented that serial provider yet. 

Thanks, Robert
